src/StructuredData.ts

```diff
@@ -96,8 +131,169 @@ export default class StructuredData {
     throw new Error("Function not implemented.")
   }
 
+  /**
+   * Converts the StructuredData back to YAML format string
+   *
+   * This method serializes the internal data structure back into valid YAML format.
+   * It only works on data that was originally loaded from YAML format to maintain
+   * data integrity and preserve the original structure.
+   *
+   * Features:
+   * - Preserves data types (strings, numbers, booleans, null)
+   * - Maintains proper YAML indentation (2 spaces)
+   * - Handles nested objects and arrays correctly
+   * - Quotes strings that could be ambiguous (look like numbers, booleans, etc.)
+   * - Supports empty objects and arrays
+   * - Ensures round-trip compatibility
+   *
+   * @returns A valid YAML string representation of the data
+   * @throws {Error} When called on data that was not originally in YAML format
+   *
+   * @example
+   * ```typescript
+   * // Load YAML data
+   * const data = yaml.from(`
+   *   name: John Doe
+   *   age: 30
+   *   hobbies:
+   *     - reading
+   *     - coding
+   * `);
+   *
+   * // Convert back to YAML
+   * const yamlString = data.toYaml();
+   * console.log(yamlString);
+   * // Output:
+   * // name: John Doe
+   * // age: 30
+   * // hobbies:
+   * //   - reading
+   * //   - coding
+   *
+   * // Verify round-trip conversion
+   * const roundTrip = yaml.from(yamlString);
+   * assert.deepStrictEqual(data.data, roundTrip.data); // true
+   * ```
+   *
+   * @example
+   * ```typescript
+   * // Error case - non-YAML data
+   * const jsonData = new StructuredData({name: "John"}, "json");
+   * jsonData.toYaml(); // Throws: Cannot convert to YAML: data was not originally in YAML format
+   * ```
+   */
   toYaml(): string {
-    throw new Error("Function not implemented.")
+    if (this.originFormat !== "yaml") {
+      throw new Error("Cannot convert to YAML: data was not originally in YAML format")
+    }
```

Comment:
This shouldn't be here. You should be able to convert data from other formats to yaml.
eg: (data that was originally in json)
```json
{ "name": "John", "age": 3, "hobbies": [ "coding", "gardening" ]}
```

should be converted into
```yml
name: John
age: 3
hobbies:
  - coding
  - gardening
```

Since there are so many formats you can skip handling all these. However remove this as it contradicts the purpose of the library